src/ORM/DataList.php

```diff
@@ -1458,8 +1517,29 @@ private function fillEmptyEagerLoadedRelations(
     public function eagerLoad(...$relationChains): static
     {
         $list = clone $this;
-        foreach ($relationChains as $relationChain) {
-            // Don't add any relations we've added before
+
+        // If an array is passed in directly, treat it as though $relationChains wasn't spread.
+        if (count($relationChains) === 1 && is_array($relationChains[array_key_first($relationChains)])) {
+            $relationChains = $relationChains[array_key_first($relationChains)];
+        }
+
+        foreach ($relationChains as $relationChain => $callback) {
+            // Allow non-associative arrays
+            if (is_numeric($relationChain)) {
+                $relationChain = $callback;
+                $callback = null;
+            }
+
+            // Reject non-callable in associative array
+            if ($callback !== null && !is_callable($callback)) {
+                throw new LogicException(
+                    'Value of associative array must be a callable.'
+                    . 'If you don\'t want to pre-filter the list, use an indexed array.'
+                );
+            }
```

Comment:
Bit of added type safety - it would cause an error later on, but throwing out a clear message about what's wrong here will improve the debugging experience